packages/project-access/src/project/search.ts

```diff
@@ -227,3 +263,145 @@ export async function findAllApps(wsFolders: WorkspaceFolder[] | string[] | unde
+            if (manifest['sap.app'] && manifest['sap.app'].type === 'library') {
+                const packageJsonPath = await findFileUp(FileName.Package, dirname(manifestPath));
+                const projectRoot = packageJsonPath ? dirname(packageJsonPath) : null;
+                if (projectRoot && (await fileExists(join(projectRoot, FileName.Ui5Yaml)))) {
+                    results.push({ projectRoot, manifestPath, manifest });
+                }
+            }
+        } catch {
+            // ignore exceptions for invalid manifests
+        }
+    }
+    return results;
+}
+
+/**
+ * Get the files to search for according to requested artifact type.
+ *
+ * @param artifacts - requests artifacts like apps, adaptations, extensions
+ * @returns - array of filenames to search for
+ */
+function getFilterFileNames(artifacts: FioriArtifactTypes[]): string[] {
+    const uniqueFilterFiles = new Set<string>();
+    for (const artifact of artifacts) {
+        if (filterFileMap[artifact]) {
+            uniqueFilterFiles.add(filterFileMap[artifact]);
+        }
+    }
+    return Array.from(uniqueFilterFiles);
+}
+/**
+ * Find all requested Fiori artifacts like apps, adaptations, extensions, that are supported by Fiori tools, for a given list of roots (workspace folders).
+ *
+ * @param options - find options
+ * @param options.wsFolders - list of roots, either as vscode WorkspaceFolder[] or array of paths
+ * @param options.artifacts - list of artifacts to search for: 'application', 'adaptation', 'extension' see FioriArtifactTypes
+ * @returns - data structure containing the search results, for app e.g. as path to app plus files already parsed, e.g. manifest.json
+ */
+export async function findFioriArtifacts(options: {
+    wsFolders?: WorkspaceFolder[] | string[];
+    artifacts: FioriArtifactTypes[];
+}): Promise<FoundFioriArtifacts> {
+    const results: FoundFioriArtifacts = {};
+    const fileNames: string[] = getFilterFileNames(options.artifacts);
+    const wsRoots = wsFoldersToRootPaths(options.wsFolders);
+    const pathMap: FileMapAndCache = {};
+    for (const root of wsRoots) {
+        try {
+            const foundFiles = await findBy({
+                fileNames,
+                root,
+                excludeFolders
+            });
+            foundFiles.forEach((path) => (pathMap[path] = null));
+        } catch {
+            // ignore exceptions during find
+        }
+    }
+    if (options.artifacts.includes('applications')) {
+        results.applications = await filterApplications(pathMap);
+    }
+    if (options.artifacts.includes('adaptations')) {
+        results.adaptations = await filterAdaptations(pathMap);
+    }
+    if (options.artifacts.includes('extensions')) {
+        results.extensions = await filterExtensions(pathMap);
+    }
+    if (options.artifacts.includes('libraries')) {
+        results.libraries = await filterLibraries(pathMap);
+    }
```

Comment:
May want to run these searches in parallel using ` Promise.all()` or similar.

Reply:
@devinea 
While I did this in some other places for HTTP request and could significantly gain performance by running requests in parallel, I've tried it for the search for apps by running the search in each workspace root parallel. I couldn't measure significant performance improvements, my assumption is that parallelized disk access doesn't really help as the disk is the single resources and access is sequential. Do you have different experience? I can easily change the code.

Reply:
@Klaus-Keller I don't have experience on parallel disk access. It just looked like code that could run in parallel.
But if there is no obvious benefit then I am fine with it as is. Cheers